grub-core/lib/gpt.c

```diff
@@ -626,10 +626,17 @@ grub_gpt_write_table (grub_disk_t disk, grub_gpt_t gpt,
 		       sizeof (*header));
 
   addr = grub_gpt_sector_to_addr (gpt, grub_le_to_cpu64 (header->header_lba));
+  if (addr == 0)
+    return grub_error (GRUB_ERR_BUG,
+		       "Refusing to write GPT header to address 0x0");
   if (grub_disk_write (disk, addr, 0, sizeof (*header), header))
     return grub_errno;
 
   addr = grub_gpt_sector_to_addr (gpt, grub_le_to_cpu64 (header->partitions));
+  if (addr < 2)
```

Comment:
The MBR is sector 0, the Primary GPT is sector 1. So the first sector that could ever be used is 2. Granted I'm checking the grub disk address which is always in units of 512 but the sector size might be larger than that but for a fool-proof check 2 is still fine.